peers/peerpool.go

```diff
@@ -334,3 +345,40 @@ func (p *PeerPool) Stop() {
 	p.serverSubscription.Unsubscribe()
 	p.wg.Wait()
 }
+
+type updateTopicRequest struct {
+	Topic  string
+	Limits params.Limits
+}
+
+// UpdateTopic updates the pre-existing TopicPool limits.
+func (p *PeerPool) UpdateTopic(topic string, limits params.Limits) error {
+	if _, err := p.getTopic(topic); err != nil {
+		return err
+	}
+
+	p.updateTopic <- &updateTopicRequest{
+		Topic:  topic,
+		Limits: limits,
+	}
+
+	return nil
+}
+
+func (p *PeerPool) updateTopicLimits(server *p2p.Server, req *updateTopicRequest) error {
+	t, err := p.getTopic(req.Topic)
+	if err != nil {
+		return err
+	}
+	t.SetLimits(req.Limits)
+	return nil
+}
+
+func (p *PeerPool) getTopic(topic string) (TopicPoolInterface, error) {
+	for _, t := range p.topics {
+		if t.Topic() == discv5.Topic(topic) {
+			return t, nil
+		}
+	}
+	return nil, errors.New("Topic not found")
```

Comment:
Errors should start with lowercase so that they can be nicely joined together using `fmt.Errorf("some error occured: %v", err)`.